### DISCUSSION TITLE
Max downloads per account

### DISCUSSION THREAD

**Discussion by Stoufs** [rejetto/hfs#422] [General]
Can you implement a setting for max downloads per 30 days per account (say 1 terabytes etc)

**Comment by rejetto** [comment]
i'm considering making this as a plugin.
I'll let you choose the amount of time and bytes, of course.

**Comment by Stoufs** [comment]
thats great!

**Comment by Stoufs** [comment]
Whats the best way to support this project? You've done a lot for me lately! And how long would it take to implement this plugin?

**Comment by rejetto** [comment]
https://github.com/rejetto/hfs#contribute

i will also invite people to write guides on the wiki, or anything that helps other users.
Of course there is also donation (the heart button), but this is not my job and money comes second, honestly. I'm in it just to make a good tool for whoever needs it.

this plugin may take one hour of work, more or less.

**Reply by Stoufs** [reply]
You are the best! I want to make this project the best it can be and be your best supporter! Its really great tool! Thank you so much for it! I look forward to testing it!

**Reply by Stoufs** [reply]
I'll be here when its done to test it immediately for you!

**Reply by rejetto** [reply]
i'll keep you posted, here, but i cannot say when. As soon as I find some free time

**Comment by rejetto** [comment]
problem: HFS has the "zip" feature. It works that it tries to determine the final size of the zip you are getting (you decide for how long), but if there are many files then the download will start anyway (when the time you decided is up).
What if the user has 100MB left, and he clicks "zip" on a folder with many files and it turns out to be 300MB in the end?

**Comment by Stoufs** [comment]
Can the plugin deny the zipped folder then if it will go over their limit?

**Comment by rejetto** [comment]
you mean, the download starts, and it is interrupted in the middle?

**Reply by rejetto** [reply]
but you could also set a lot of seconds for the zip, so that you are quite sure that you'll know the size in advance
<img width="252" alt="image" src="https://github.com/rejetto/hfs/assets/1367199/6496aa39-9768-4bac-9a3b-6a22f5416f36">

**Reply by Stoufs** [reply]
Would is still be possible to ensure no matter what that the download is interrupted if it goes over?

**Comment by rejetto** [comment]
yes, i can work on that, but if it is interrupted in the middle there's no way (currently) to show a message and explain why

**Reply by Stoufs** [reply]
That's fine. No message is needed

**Comment by rejetto** [comment]
the counting of the days/time can be a tough problem.
should it be the same for all accounts, or should it start for each account the moment they make their first download?

**Reply by Stoufs** [reply]
yea first download is fine

**Reply by Stoufs** [reply]
the days / time im not worried about as i will be tracking that manually. just the total amount of bytes downloaded limit mainly.

**Comment by rejetto** [comment]
so, anonymous users won't be affected. Of course you can prevent anonymous access by configuring your HFS.

**Comment by Stoufs** [comment]
Correct. Will not affect anonymous.

**Comment by rejetto** [comment]
admin > plugins > search
it's called download-quota
<img width="441" alt="image" src="https://github.com/rejetto/hfs/assets/1367199/f93abbbe-781e-428b-b66d-0f2448b85ca7">

i just had the time for a minimum amount of testing

**Reply by Stoufs** [reply]
This is excellent! Loving it! Is there a way to dispaly to the user to see what their usage is like left?

**Comment by rejetto** [comment]
technically yes, but i have to think how. Maybe the user-panel.
it is displayed in the error message, currently
`You have reached your ${formatBytes(quota)} quota, that will reset on ${formatTimestamp(expires)}`

**Reply by Stoufs** [reply]
Ya in the user panel when they click t heir name would be awesome!

**Reply by Stoufs** [reply]
Oh i see what you are talking about. I mean to dispaly how much is left. The error message only shows they went over.

**Comment by Stoufs** [comment]
Can you make it so the account is disabled after the quota reaches its peak?

**Comment by rejetto** [comment]
that is not really compatible with the logic, as the user never really reach its quota.
Example: when you are left with 10KB, and try to download files that are larger, you get the message, but you have still some left.

**Reply by rejetto** [reply]
and because of this, i guess i should choose the error message differently, telling how much is left

**Reply by Stoufs** [reply]
That would be awsome!

**Reply by rejetto** [reply]
no prob, update plugin, version 0.11 is available

**Comment by rejetto** [comment]
<img width="260" alt="image" src="https://github.com/rejetto/hfs/assets/1367199/9d334e30-4288-4dcf-9fac-7da9c5218004">

with version 0.2 of plugin, but this will be working only with next HFS 0.51

**Comment by Stoufs** [comment]
Ah ok so not available yet

**Reply by rejetto** [reply]
exactly. The plugin is, but the thing won't show up until we get HFS updated

**Reply by Stoufs** [reply]
Gotcha.  Looking forward to new update! Do you have an ETA?

**Comment by rejetto** [comment]
probably 2-4 days. But since you are in a hurry, here is a preview
[hfs-mac-arm.zip](https://github.com/rejetto/hfs/files/13796348/hfs-mac-arm.zip)
[hfs-mac.zip](https://github.com/rejetto/hfs/files/13796349/hfs-mac.zip)
[hfs-linux.zip](https://github.com/rejetto/hfs/files/13796350/hfs-linux.zip)
[hfs-windows.zip](https://github.com/rejetto/hfs/files/13796351/hfs-windows.zip)

it should be possible to copy the link, then go to admin-panel > home, and right click on check-for-updates
